### DISCUSSION TITLE
Define equivalent weights for fields marked as searchableAttribute

### DISCUSSION THREAD

**Discussion by gmourier** [meilisearch/product#512] [Feedback & Feature Proposal]
Hey 👋,

Currently, the fields described in the `searchableAttributes` settings cannot have the same weight.

1. If `*` is defined (this is the default value), fields are weighted by their position in the first indexed document. So if `title` appears before `description`, `title` will be more important at search time.
2. If fields are defined, they can't have the same weight; The more the field is located on the left side of the array the more important it is and so on.

Meilisearch could incorporate the ability to have fields that weigh exactly the same to gain versatility in search results relevancy tweaking.

e.g. 

```json
{
     "searchableAttributes": [
               "title", 
               ["description", "detailed_description"]
     ]
}
```

In this example `title` is more important than `description` and `detailed_description` but `description` and `detailed_description` have the exact same weight.

Would you be interested in seeing this capability introduced to Meilisearch? We would like to know more about your needs.

Leave a comment explaining your use case!

Thank you 🙏

**Comment by muayyad-alsadi** [comment]
since title is very short and description might contain irrelevant keywords (title: used playstation, description: find our shop next to toyota dealer)

we might want to give 10x multiplier to the weight of title keywords something like this (or avg. length of desc / avg. length of title )

```
{
    "searchableAttributesWithWeights": {
            "title": 10, 
             "description":1, "detailed_description":1,
    }
}
```
